### PR TITLE
 Fibonacci series 

### PR DESCRIPTION
**Pull request by Vinayakk404**
## Related Issue
- Adding Fibonacci series using dynamic programming
- Linked issue #101 

## Additional information
- Created a cpp program in the folder dynamic programming that generates Fibonacci series (0,1,1,2...... and so on) up to the nth term (n is an integer input given by the user). 
- Required and relevant comments have also been added in the cpp file to assist anyone reading the code along with space and time complexity

## Proposed Changes
- added Fibonacci series.cpp

## Checklist

- [x] ✅ My code follows the code style of this project.
- [ ] 📝 My change requires a change to the documentation.
- [ ] 🎀 I have updated the documentation accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- Fibonacci series contain numbers that are the sum of two preceding numbers
example = 0,1,1,3,4.............. and so on
- In this code we have tried to print these series using Dynamic programming  .
- This code prints Fibonacci series upto nth terms(n is a positive integer provided by the user)
- We use dynamic programming here as it reduces the Time complexity drastically as compared to recursion

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
Screenshot #1    
Test 1 where n=5  
 image

Screenshot #2  
Test 2 where n=10   
image

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
Hey @Vinayakk404  , make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.

**Comment by n4i9kita** [comment]
@Vinayakk404, please don't make so many commits per issue. 
Take your time.
But, next time, avoid making so many commits (with such minor changes)

**Comment by n4i9kita** [comment]
@Vinayakk404, this CPP file will go in Recursion folder.
Why dynamic programming folder?

**Review by n4i9kita — COMMENTED** [review]
Write a better readme. Or else, let some else do that task. 
For reference, this is how you are supposed to write -> [here](https://github.com/Lakhankumawat/LearnCPP/tree/main/S-SortingAlgorithms).

Properly Organized. 
Avoid using HTML tags in Markdown.

**Review by n4i9kita — COMMENTED** [review]
@Vinayakk404 

Feedback : 
1. Indent your code properly.
2. Write better comments ("making the first term zero" -> this part was self-explanatory)
3. Please help me understand why you have declared array F of size (n+1)  [line no. 11]? Let's say n=1, means we want to print the first element. Can't it be done by F[0], why do we need F[1] ?
4. Put a comma between the element values rather than space.

**Comment by Vinayakk404** [comment]
> @Vinayakk404
> 
> Feedback :
> 
> 1. Indent your code properly.
> 2. Write better comments ("making the first term zero" -> this part was self-explanatory)
> 3. Please help me understand why you have declared array F of size (n+1)  [line no. 11]? Let's say n=1, means we want to print the first element. Can't it be done by F[0], why do we need F[1] ?
> 4. Put a comma between the element values rather than space.

Thank you for for taking some time and having patience for going through and provide feedback for my code.
 I have indented the code , tried to write better comments , corrected my mistake and put comma between the output numbers.
In future i will try to keep commits to a minimum

File: D-DynamicProgramming/Fibonacci_series.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  else if(n==2) {
+     cout<<F[0]<<","<<F[1];  //printing the first two terms
+      
+  } 
+
+  else {
+
+     cout<<F[0]<<","<<F[1];
+
+  for(int i=2;i<n;i++){
```
Comment (n4i9kita) [inline D-DynamicProgramming/Fibonacci_series.cpp:27]: Not indented.

File: D-DynamicProgramming/Fibonacci_series.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     F[i]=F[i-1]+F[i-2];  // adding preceeding two terms to get the next term
+     cout<<","<<F[i]; 
+    }
+  }
+}
+
+ //Driver meathod
```
Comment (n4i9kita) [inline D-DynamicProgramming/Fibonacci_series.cpp:34]: Spelling of `method`.

File: D-DynamicProgramming/Fibonacci_series.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+int main()
+{
+   int n;
+   cout<<"Enter the value of n"<<endl;
+   cin>>n;
+   if (n<1){              //as the series starts from 1st term
```
Comment (n4i9kita) [inline D-DynamicProgramming/Fibonacci_series.cpp:40]: This comment can be removed.

File: D-DynamicProgramming/Fibonacci_series.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+   } 
+   else{
+     cout<<"Required fibonacci series is ";
+     fibonacci(n);
+
+   }
+    return 0;
+}
+
+//The space complexity is O(n)
```
Comment (n4i9kita) [inline D-DynamicProgramming/Fibonacci_series.cpp:51]: The space and time complexity will be added in readme.

File: D-DynamicProgramming/Fibonacci_series.cpp (hunk 1 above)
Comment (Vinayakk404) [reply D-DynamicProgramming/Fibonacci_series.cpp]: Done

File: D-DynamicProgramming/Fibonacci_series.cpp (hunk 2 above)
Comment (Vinayakk404) [reply D-DynamicProgramming/Fibonacci_series.cpp]: Corrected

File: D-DynamicProgramming/Fibonacci_series.cpp (hunk 3 above)
Comment (Vinayakk404) [reply D-DynamicProgramming/Fibonacci_series.cpp]: Removed

File: D-DynamicProgramming/Fibonacci_series.cpp (hunk 4 above)
Comment (Vinayakk404) [reply D-DynamicProgramming/Fibonacci_series.cpp]: Removed that comment from .cpp file

**Comment by n4i9kita** [comment]
LGTM!

**Review by miraj0507 — COMMENTED** [review]
Code looks good to me.

**Comment by Lakhankumawat** [comment]
@Vinayakk404 i would recommend you to fetch upstream and recreate the Pull Request with same changes and i will accept it because for now there exists some conflicts

**Comment by Lakhankumawat** [comment]
Just tag me in discord with This PR name so that i can directly merge your PR and mentors will not have to review it again

**Comment by Lakhankumawat** [comment]
There are conflicts

**Comment by Vinayakk404** [comment]
@Lakhankumawat can I redo it and create another pull request

**Comment by Lakhankumawat** [comment]
If you created a new fresh PR with all the changes mentioned then just tag me in discord and mention i reviewed your old pr as well as in new pr mention old pr number !

**Pull request closed without merging**